packages-next/fields/src/types/integer/index.ts

```diff
@@ -30,7 +30,7 @@ export const integer =
   meta =>
     fieldType({
       kind: 'scalar',
-      mode: 'optional',
+      mode: isRequired ? 'required' : 'optional',
```

Comment:
We must match the `isRequired` value to set correct mode, this is needed to check that idField is required. Maybe other fields need the same fix, but this is outside of current PR.